### PR TITLE
feat: implement or operator

### LINKED ISSUE #10054: Nested "OR" Condition/Operation Support in Repository Where condition
### Feature Description

I would like to achieve the below using repository.where method

Select * from user where (firstName like "%Mike% **OR** firstName IN ("King", "Kong")) AND (lastName like "%Mike% **OR** lastName IN ("King", "Kong"))

I am aware this can be acheived through query builder and raw queries but currently there seems to no way to achieve this using the repository.where method.  Is it a good idea to introduce a OR operator similar to the existing AND operator to achieve the above. 

### The Solution

Introduce OR operator similar to the AND operator to do something similar like below. 

await repository.find( 
  where: {
    firstName: **OR**(Like('%Mike%'), In('King','Kong')),
    lastName: **OR**(Like('%Mike%'), In('King','Kong')),
 },
})

### Considered Alternatives

This can be achieved through query builder and raw queries.

### Additional Context

_No response_

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [ ] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [X] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

### PR DESCRIPTION
**Pull request by deepkhurana1999**
This new feature enables typeorm to allow use multiple operators joining with or

Closes: #10054

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run format` to apply prettier formatting
- [ ] `npm run test` passes with this change
- [x] This pull request links relevant issues as `Fixes #10054`
- [x] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by deepkhurana1999** [comment]
Note: The implementation is tested with the new test-case.

The following test-cases are failed custom cache provider on the feature as well as master branch. Therefore, not able to run the complete test

File: src/index.ts [hunk 1]
```diff
@@ -67,6 +67,7 @@ export * from "./decorator/Exclusion"
 export * from "./decorator/Generated"
 export * from "./decorator/EntityRepository"
 export * from "./find-options/operator/And"
+export * from "./find-options/operator/OR"
```
Comment (soomtong) [inline src/index.ts:70]: It seems to be 'OR'

Comment (deepkhurana1999) [reply src/index.ts]: Ok, I have updated it

**Review by deepkhurana1999 — COMMENTED** [review]
Ok, I have renamed it.

**Comment by deepkhurana1999** [comment]
> Ok, I have renamed it.

[Update:] Seems like visual studio code not detecting file name changes. Therefore, removed and adding it again.

**Comment by deepkhurana1999** [comment]
@soomtong Any update?

**Comment by soomtong** [comment]
> @soomtong Any update regarding review?

It would be great if you could also include the 'OR' in the readme.md document. 
I'm interested in your PR and wanted to give some input, but other maintainers need to review it.
Anyway I like this implementation!

**Comment by deepkhurana1999** [comment]
> > @soomtong Any update regarding review?
> 
> It would be great if you could also include the 'OR' to 'Or' in the readme.md document. I'm interested in your PR and wanted to give some input, but other maintainers need to review it. Anyway I like this implementation!

Thank you @soomtong. 

There is no dedicated section to explain **FindOptions** in Readme.md, I have already updated the find-options.md. Please let me know should I create a new section about FindOptions?

File: docs/find-options.md [hunk 2]
```diff
@@ -622,3 +624,19 @@ will execute following query:
 ```sql
 SELECT * FROM "post" WHERE NOT("likes" > 10) AND NOT("title" = 'About #2')
 ```
+
+-   `OR`
```
Comment (soomtong) [inline docs/find-options.md:628]:
```suggestion
-   `Or`
```

File: docs/find-options.md [hunk 3]
```diff
@@ -622,3 +624,19 @@ will execute following query:
...
+
+```ts
+import { Not, MoreThan, ILike } from "typeorm"
+
+const loadedPosts = await dataSource.getRepository(Post).findBy({
+    title: OR(Equal("About #2"), ILike("About%")),
```
Comment (soomtong) [inline docs/find-options.md:634]:
```suggestion
    title: Or(Equal("About #2"), ILike("About%")),
```

File: src/find-options/operator/Or.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { FindOperator } from "../FindOperator"
+
+export function OR<T>(...values: FindOperator<T>[]): FindOperator<T> {
```
Comment (soomtong) [inline src/find-options/operator/Or.ts:3]:
```suggestion
export function Or<T>(...values: FindOperator<T>[]): FindOperator<T> {
```

File: test/github-issues/10054/issue-10054.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import { OR, DataSource, ILike, Equal } from "../../../src"
```
Comment (soomtong) [inline test/github-issues/10054/issue-10054.ts:1]:
```suggestion
import { Or, DataSource, ILike, Equal } from "../../../src"
```

File: test/github-issues/10054/issue-10054.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import {
+    closeTestingConnections,
+    createTestingConnections,
+    reloadTestingDatabases,
+} from "../../utils/test-utils"
+import { Person } from "./entity/Person"
+import { expect } from "chai"
+
+describe("github issues > #10054 Nested 'OR' Condition/Operation Support in Repository Where condition", () => {
```
Comment (soomtong) [inline test/github-issues/10054/issue-10054.ts:10]:
```suggestion
describe("github issues > #10054 Nested 'Or' Condition/Operation Support in Repository Where condition", () => {
```

File: test/github-issues/10054/issue-10054.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                dave.age = 12
+
+                const foobar = new Person()
+                foobar.name = "FooBar"
+                foobar.age = 14
+
+                await dataSource.manager.save([foo, john, dave, foobar])
+                const persons = await dataSource.manager.find(Person, {
+                    where: {
+                        name: OR(ILike("foo%"), Equal("Jane")),
```
Comment (soomtong) [inline test/github-issues/10054/issue-10054.ts:48]:
```suggestion
                        name: Or(ILike("foo%"), Equal("Jane")),
```

**Review by soomtong — COMMENTED** [review]
This is what I found, and I hope it helps.

File: test/github-issues/10054/issue-10054.ts (hunk 7 above)
Comment (deepkhurana1999) [reply test/github-issues/10054/issue-10054.ts]: > This is what I found, and I hope it helps.

Thank you, I have updated as requested.

**Comment by deepkhurana1999** [comment]
> This is what I found, and I hope it helps.

Hello @soomtong,

I have addressed all of the changes you requested. Please take a look and let me know if you have any questions or concerns.

File: test/github-issues/10054/issue-10054.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+                dave.age = 12
+
+                const foobar = new Person()
+                foobar.name = "FooBar"
+                foobar.age = 14
+
+                await dataSource.manager.save([foo, john, dave, foobar])
+                const persons = await dataSource.manager.find(Person, {
+                    where: {
+                        name: Or(ILike("foo%"), Equal("Jane")),
```
Comment (soomtong) [inline test/github-issues/10054/issue-10054.ts:48]: I hope this patch gets merged soon, it would be great for casual users.

**Review by soomtong — COMMENTED** [review]
LGTM

**Comment by deepkhurana1999** [comment]
Hello @pleerock,

@soomtong has reviewed the PR and everything seems to looking good. Please take a look into the PR and let me know if you have any questions or concerns.

**Comment by benjefferies** [comment]
Top work @deepkhurana1999. Would be great to see this in

**Review by hinogi — APPROVED** [review]
LGTM

**Comment by deepkhurana1999** [comment]
> LGTM

Thank you, @soomtong @hinogi any plan on merging the changes?

**Comment by hinogi** [comment]
> > LGTM
> 
> Thank you, @soomtong @hinogi any plan on merging the changes?

Sorry, I am unable to do such a thing, I am not a maintainer.

**Comment by pleerock** [comment]
sorry for late review here and thank you for your work.

**Pull request merged**